src/west/app/project.py

```diff
@@ -795,6 +795,11 @@ def do_add_parser(self, parser_adder):
                            help='''proceed as if FILTER was appended to
                            manifest.group-filter; may be given multiple
                            times''')
+        group.add_argument('--submodule-init-config',
```

Comment:
for the convenience of users who need to do this, should we support this on a west workspace basis.
For example by supporting a:
```
$ west config update.submodule-init-config protocol.file.allow=always
```
setting or similar.

That would allow users to set this on a per-workspace instead of globally / having to repeat on each update command.

Such proposal can be made on a followup PR.